### DISCUSSION TITLE
New version Now Available 🎉

### DISCUSSION THREAD

**Discussion by BigBobbas** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#90] [Announcements]
I am just finishing the final stages of the next revision of the firmware. Currently grabbing screenshots to update the documentation and making a few final tweaks and tidying up. Then this will be merged straight to the main branch. I will create an archive branch with all old configs etc but will close the dev and wip branches once everything is in place.

That will be it for the version, using the standard display component, I am then going to take some time out to work on a few of my own projects . I will still be around to answer questions and to provide support and squash any bugs etc.

I will be posting updates on other stuff that I will be working on [discord](<https://discord.gg/U3SRYCG6Kp>) and will be around for a chat . 
I'm then going to start working on an LVGL version of the firmware for the ESP32 - S3 - Box- 3 , which will hopefully make things a little simpler to customise  and the features that LVGL provide are pretty awesome.

I will add an update once everything is in place and good to go. This will be the firmware that is also flashed when using the web installer.

I would also like to say a big thank you to all of you fantastic people for getting involved and supporting this project, whether you have bought me a coffee, installed the firmware  , contributed, or pointed out any issues you are all appreciated and the reason why I do this it's  a fun project and a pleasure to work on. so THANK YOU! :heart:  and keep the suggestions and feedback coming for 'Round 2' when we give the box a 'make over'

Best Wishes 
Rob 

## UPDATE
ok the latest version is now live - available to install from https://support.bbdl.co.uk/ for the version with the sensor dock  or manually  [here](<https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/main/s3b.yaml>)
For the version without a sensor Dock [here](<https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/main/s3b_no_sensor.yaml>)
Please let me know if you have any issues preferably on [Discord](<https://discord.gg/U3SRYCG6Kp>). If you have made customisations then please make sure to copy the relevant sections of config into the new one. And make sure to make a fully copy of your config before updating.
**** Updated docs to follow ****

**Comment by damanic** [comment]
Just compiled and installed `s3b_import.yaml
` from command line with latest esphome. 

A couple of issues:

- The alarmo button works but it is missing the icon in top bar (appears not to be there)
- The timer works great except there is no audible noise when countdown complete. I hear a faint clicking noise that is all.

**Reply by BigBobbas** [reply]
Hi ,
i'm in the middle of updating instructions etc.. the 2 sound files from here https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/tree/Archive/sounds need to be in config/www/sounds if they aren't already.
 i will flash and double check if its an issue with the config too.
 i'll also look into the alrmo icon this usually appears once HA connects to the API.
 I'll update you shortly and thanks for letting me know 👍

**Comment by BigBobbas** [comment]
I have tested and both the timer sound and Alarmo icon are appearing. So it looks like all is good with the config. 
In the config under the `logger:` component can you comment out the line `text_sensor: WARN` this should be around line 132
If you then install and watch the logs, you should see a text_sensor entry for Alarmo can you check to see what state it shows.
Cheers
Rob

**Reply by Tobster77** [reply]
@BigBobbas I now observe the same issue. I use Alarmo, but the icon is not visible. I commented out the the line `text_sensor: WARN` and I see only a single occurrence of the word "alarmo" directly on boot:

`[15:10:11][C][homeassistant.text_sensor:024]:   Entity ID: 'alarm_control_panel.alarmo'`

After this, the word "alarmo" does not come up again in the logs. I also switched the alarmo alarm on and off, but this does not lead to a reaction. Is there something I have to configure in the s3b.yaml file for Alarmo? An alarm area or something? This might differ between your and my installation.

I also saw that you employ an entity `alarm_control_panel.alarmo` while in my HA entities, I could find a panel per alarm zone in my house, including an automatically created `alarm_control_panel.master`, but not `alarm_control_panel.alarmo`. Does this play a role?

**Reply by BigBobbas** [reply]
Hi there ,
to firstly answer your other question. yes LVGL is still going to happen. I have been learning the basics of it so I can apply this to a new firmware so hopefully early in 2025.

With regards to Alarmo, the config will look in HA for that entity `alarm_control_panel.alarmo` which should be installed as part of the Alarmo integration. 

If you open the Alarmo integration, you should see 
![image](https://github.com/user-attachments/assets/812565e0-e339-43bb-8b0e-3fbeaf588076)

click the icon then click on the 'cog' icon in the top right corner and it should show the entity.
![image](https://github.com/user-attachments/assets/3433ffdd-5fc8-47be-bb21-e13bfa19b083)

**Reply by Tobster77** [reply]
Hi,

great to hear about LVGL. This hopefully allows for more simple modularization in order to make customization of main screen and pages simpler, right?

Indeed, this looks different for me:
![image](https://github.com/user-attachments/assets/6333a6d4-e76f-473a-bfd2-4fd7630f79a9)
![image](https://github.com/user-attachments/assets/66d3052f-de1f-4c16-9106-89b6d1b2ba2f)


To my limited knowledge, Alarmo uses different zones, where in your case, only one zone called Alarmo was created. Now in any case, Alarmo has a setting where it asks to create a master zone covering all zones, which is enabled for me. This setting created the master zone.

I tested to replace `alarm_control_panel.alarmo` by `alarm_control_panel.master` and this worked perfectly. I now see the icon and also get a screen to see and set the alarm mode.

So one would have to set a zone there, I think. Could this be a variable?

**Reply by BigBobbas** [reply]
Yes , I have just looked at this now and can see the difference
I will have to spend sometime looking into this as I have only used the one area/zone and used different actions for arming.

this is the first time that this has been mentioned so I was unaware.

**Comment by damanic** [comment]
OK my bad, I didn't copy the timer mp3 to home assistant.   Sound issue resolved.

I could not resolve the icon issue for Alarmo. I tried:

```
rm -rf .esphome
rm -rf build
esphome compile s3b_import.yaml
```

Then I realised that my alarm set up uses the alarm helper control panel -> https://www.home-assistant.io/integrations/alarm_control_panel/   .  Not Alarmo.  So perhaps the icon not being shown is intended in this case?

I am not sure if Alarmo uses HA's native helper for arm/disarm. If it does it might be better to tie the alarm feature to the helper instead of Alarmo so that it works for both. Otherwise a note/switch in yaml config to choose Alarmo or HA's `alarm_control_panel` helper for the arm/disarm actions.     

Anyway, that suggestion aside,  I can confirm this latest release compiles and appears to work as intended on my S3 - great work - thanks!

**Comment by BigBobbas** [comment]
Hi there 
thank you for your feedback. The implementation is for Alarmo only as this is a very commonly used alarm control integration, I have never used the HA alarm panel so not familiar wit how it functions. I will have a look into it but I am not currently looking to add any more features to this version as the next version will be to use LVGL for the display UI.  Any new features would be added to that version .. which will not be for some time as I have have other projects of my own to do first.
However if anyone does want to write the config for HA alarm and submit a PR I would definitley check it out.
Cheers
Rob

**Comment by damanic** [comment]
@BigBobbas 
Perhaps when you are done with this version update you could tag it and release so that we can keep track of any minor updates and for reference should you decide to overhaul the repo to push v2 LVGL releases in future.

**Reply by BigBobbas** [reply]
cheers - i am not experienced with github .. or any of this and am still learning. I will certainly look at this and put it in place, thanks for the pointer 👍

**Comment by Tobster77** [comment]
@BigBobbas This LVGL initiative sounds awesome - is this still on the list? Would be happy to join testing :-)

And the integration of Alarmo is quite what I had to do manually so far - thank you! Currently compiling...

**Comment by Tobster77** [comment]
Hi,

thanks to @BigBobbas for all this fantastic work. I'm thinking about some adaptions... would it make sense to wait for the LVGL version?

**Reply by BigBobbas** [reply]
TBH at the moment I haven't the time to do a full on LVGL version. I can confirm that everything will work on the S3 box using LVGL. but due to the time it would require to do a config that people could easily adapt would take quite a considerable time.

There are a few examples of configs in our discord server that have been done. It wouldn't take too much for an individual to make a config to suit their own needs as you wouldn't need to factor in all the "what if's" required to make a universal config. 

What kind of things did you have in mind? we can always see what would be the best way to implement them 👍
